File: server/job/src/main/java/io/iamcyw/ams/job/strategy/persistence/StrategyLevelTargetPO.java
package io.iamcyw.ams.job.strategy.persistence;

import io.quarkus.hibernate.orm.panache.PanacheEntity;

import javax.persistence.*;

@Entity(name = "ALARM_STRATEGY_LEVEL_TARGET")
public class StrategyLevelTargetPO extends PanacheEntity {
    @Id
    @GeneratedValue
    public Long id;

    @ManyToOne
    @JoinColumn(name = "LEVEL_ID", nullable = false)
    private StrategyLevelPO strategyLevel;

    private TargetType targetType;

    private String value;

    enum TargetType {
        User, UserGroup, Address
    }

}

File: domain/src/commonMain/kotlin/io/iamcyw/ams/domain/job/strategy/StrategyLevelTargetPO.kt
package io.iamcyw.ams.domain.job.strategy

expect class StrategyLevelTargetPO {

    var strategyLevel: StrategyLevelPO?

    var type: TargetType?

    var vale: String

}

enum class TargetType {
    User, UserGroup, Address
}